sdcm/nemesis.py

```diff
@@ -379,19 +386,43 @@ class UpgradeNemesis(Nemesis):
 
     # upgrade a single node
     def upgrade_node(self, node):
+        repo_file = self.cluster.params.get('repo_file', None,  'scylla.repo.upgrade')
+        new_version = self.cluster.params.get('new_version', None,  '')
+        upgrade_node_packages = self.cluster.params.get('upgrade_node_packages')
         self.log.info('Upgrading a Node')
+
+        # We assume that if update_db_packages is not empty we install packages from there.
+        # In this case we don't use upgrade based on repo_file(ignored sudo yum update scylla...)
         orig_ver = node.remoter.run('rpm -qa scylla-server')
-        scylla_repo = get_data_path('scylla.repo.upgrade')
-        node.remoter.send_files(scylla_repo, '/tmp/scylla.repo', verbose=True)
-        node.remoter.run('sudo cp /etc/yum.repos.d/scylla.repo ~/scylla.repo-backup')
-        node.remoter.run('sudo cp /tmp/scylla.repo /etc/yum.repos.d/scylla.repo')
-        # backup the data
-        node.remoter.run('sudo cp /etc/scylla/scylla.yaml /etc/scylla/scylla.yaml-backup')
-        node.remoter.run('sudo nodetool snapshot')
-        node.remoter.run('sudo chown root.root /etc/yum.repos.d/scylla.repo')
-        node.remoter.run('sudo chmod 644 /etc/yum.repos.d/scylla.repo')
-        node.remoter.run('sudo yum clean all')
-        node.remoter.run('sudo yum update scylla scylla-server scylla-jmx scylla-tools scylla-conf scylla-kernel-conf -y')
+        if upgrade_node_packages:
+            # update_scylla_packages
+            node.remoter.send_files(upgrade_node_packages, '/tmp/scylla', verbose=True)
+            # node.remoter.run('sudo yum update -y --skip-broken', connect_timeout=900)
+            node.remoter.run('sudo yum install python34-PyYAML -y')
+            # replace the packages
+            node.remoter.run('yum list installed | grep scylla')
+            node.remoter.run('sudo nodetool snapshot')
+            # update *development* packages
+            node.remoter.run('sudo rpm -UvhR --oldpackage /tmp/scylla/*development* | true')
```

Comment:
1. Can we replace the `true` with grace (ignore_status=True)?

2. How about remove development pattern? then we can install anything.
such as some minor release that doesn't exists in repo, this provide a way to upgrade/downgrade of any minor release.   @abvgedeika @roydahan